src/docs_website/src/file_checker.zig

```diff
@@ -70,5 +70,62 @@ fn check_files(arena: std.mem.Allocator, path: []const u8) !void {
                 return error.MissingNewline;
             }
         }
+
+        if (std.mem.eql(u8, file_type.extension, ".html")) {
+            const html_path = try std.fs.path.join(arena, &.{ path, entry.path });
+            try check_links(arena, html_path);
+        }
+    }
+}
+
+// These links don't work with https.
+const https_exceptions = std.StaticStringMap(void).initComptime(.{
+    .{"http://www.bailis.org/blog/linearizability-versus-serializability/"},
+});
+
+fn check_links(arena: std.mem.Allocator, html_path: []const u8) !void {
+    const dirname = std.fs.path.dirname(html_path).?;
+    const html = try std.fs.cwd().readFileAlloc(arena, html_path, file_size_max);
+
+    var line_it = std.mem.tokenizeScalar(u8, html, '\n');
+    var line_number: usize = 1;
+    while (line_it.next()) |line| {
+        defer line_number += 1;
+        errdefer log.err("[link checker] error in {s}:{}", .{ html_path, line_number });
+
+        var link_it = std.mem.tokenizeSequence(u8, line, "href=\"");
+        if (!std.mem.startsWith(u8, line, "href=\"")) _ = link_it.next(); // Skip prefix
+        while (link_it.next()) |chunk| {
+            var url = std.mem.sliceTo(chunk, '"');
+
+            // Strip anchor
+            if (std.mem.lastIndexOfScalar(u8, url, '#')) |i| url = url[0..i];
+            if (url.len == 0) continue;
+
+            if (std.mem.startsWith(u8, url, "http://")) {
+                if (https_exceptions.has(url)) continue;
+                log.err("Found insecure link: {s}", .{url});
+                return error.InsecureLink;
+            }
+
+            // Skip external link
+            if (std.mem.startsWith(u8, url, "https://")) continue;
```

Comment:
Maybe add external link validation under a flag (just using http.CLient to fetch and check that its 200 ok)

Reply:
Oh, and we also punctuate comments, so `// Strip external link.`:

https://github.com/tigerbeetle/tigerbeetle/blob/main/docs/TIGER_STYLE.md#naming-things